### PR TITLE
Added a note to the serialization section regarding desugared names

### PR DESCRIPTION
**Pull request by WadeWaldron**
Fixes issue https://github.com/akka/akka/issues/21263

### CODE REVIEW

File: akka-docs/rst/java/serialization.rst [hunk 1]
```diff
@@ -35,6 +35,14 @@ one of which all other candidates are superclasses. If this condition cannot be
 met, because e.g. ``java.io.Serializable`` and ``MyOwnSerializable`` both apply
 and neither is a subtype of the other, a warning will be issued.
 
+.. note::
+
+  If you are using Scala for your message protocol and your messages are contained
+  inside of a Scala wrapper object, then in order to reference those messages, you will need
```
Comment (ktoso) [inline akka-docs/rst/java/serialization.rst]: skip wrapper, just Scala `object`

File: akka-docs/rst/java/serialization.rst [hunk 2]
```diff
@@ -35,6 +35,14 @@ one of which all other candidates are superclasses. If this condition cannot be
...
+  use the de-sugared path. For a message named ``Message`` contained inside
+  the Scala object named ``Wrapper`` you would need to reference it as
+  ``Wrapper$Message`` instead of ``Wrapper.Message``
```
Comment (ktoso) [inline akka-docs/rst/java/serialization.rst]: `.` at end of sentence :)

File: akka-docs/rst/scala/serialization.rst [hunk 3]
```diff
@@ -35,6 +35,13 @@ one of which all other candidates are superclasses. If this condition cannot be
 met, because e.g. ``java.io.Serializable`` and ``MyOwnSerializable`` both apply
 and neither is a subtype of the other, a warning will be issued
 
+.. note::
+
+  If your messages are contained inside of a wrapper object, then in order to
```
Comment (ktoso) [inline akka-docs/rst/scala/serialization.rst]: inside of a Scala `object`

File: akka-docs/rst/java/serialization.rst [hunk 4]
```diff
@@ -35,6 +35,14 @@ one of which all other candidates are superclasses. If this condition cannot be
 met, because e.g. ``java.io.Serializable`` and ``MyOwnSerializable`` both apply
 and neither is a subtype of the other, a warning will be issued.
 
+.. note::
+
+  If you are using Scala for your message protocol and your messages are contained
+  inside of a Scala wrapper object, then in order to reference those messages, you will need
+  use the de-sugared path. For a message named ``Message`` contained inside
```
Comment (johanandren) [inline akka-docs/rst/java/serialization.rst]: `fully qualified Java class name`?

File: akka-docs/rst/java/serialization.rst (hunk 2 above)
Comment (ktoso) [inline akka-docs/rst/java/serialization.rst]: Perhaps add a word of explanation, "This the how the class is represented once compiled by scalac, thus we need to use this path to reach it"?

**Comment by johanandren** [comment]
LGTM

**Comment by ktoso** [comment]
Thanks :)
LGTM after minor comments

File: akka-docs/rst/java/serialization.rst (hunk 4 above)
Comment (ktoso) [inline akka-docs/rst/java/serialization.rst]: Good point

**Comment by johanandren** [comment]
Great, thanks! Needs a rebase for some reason though

**Comment by johanandren** [comment]
No it doesn't, github UI lied to me.

**Pull request merged**